### ISSUE TITLE
`tf.experimental.numpy.divmod` produces `nan` remainder for `1 % inf` on GPU

### ISSUE THREAD

**Issue by fuzzyswan** [tensorflow/tensorflow#58493]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.9.2

### Custom Code

Yes

### OS Platform and Distribution

Colab

### Python version

3.7.15

### CUDA/cuDNN version

11.2.152, 8.1.1

### Current Behaviour?

When computing the remainder `1 % inf`, both `np.divmod` and `tf.experimental.numpy.divmod` on CPU produces `1.0`, but `tf.experimental.numpy.divmod` on GPU produces `nan`.

### Standalone code to reproduce the issue

```shell
print("numpy", np.divmod(1.0, np.inf))
with tf.device("/cpu:0"):
  print("CPU", tf.experimental.numpy.divmod(1.0, np.inf))
with tf.device("/gpu:0"):
  print("GPU", tf.experimental.numpy.divmod(1.0, np.inf))
```

### Relevant log output

```shell
numpy (0.0, 1.0)
CPU (<tf.Tensor: shape=(), dtype=float64, numpy=0.0>, <tf.Tensor: shape=(), dtype=float64, numpy=1.0>)
GPU (<tf.Tensor: shape=(), dtype=float64, numpy=0.0>, <tf.Tensor: shape=(), dtype=float64, numpy=nan>)
```
</details>

**Comment by bhack** [comment]
Yes it is the underline impl of `tf.math.mod`

```python
import numpy as np
import tensorflow as tf
tf.config.set_soft_device_placement(False)
print("numpy", np.divmod(1.0, np.inf))
with tf.device("/cpu:0"):
  print(tf.math.mod(1.0,np.inf))
with tf.device("/gpu:0"):
  print(tf.math.mod(1.0,np.inf))
```

I think that it is the same as https://github.com/tensorflow/tensorflow/issues/58369

It is better to contribute a PR to extend the coverage of the underline TF op as suggested at https://github.com/tensorflow/tensorflow/issues/58496#issuecomment-1309127218.

**Comment by JW1992** [comment]
Closing it as the underlying cause should have been fixed in [#58369](https://github.com/tensorflow/tensorflow/issues/58369#issuecomment-1373394109). As a verification I cannot reproduce it using the latest version
